### PR TITLE
check_oneof function

### PR DESCRIPTION
**Pull request by lukesneeringer**
This commit adds a `google.gax.utils.oneof.check_oneof` function that raises ValueError if two or more keyword arguments are truthy.

Required for googleapis/toolkit#1069.

### CODE REVIEW

**Comment by lukesneeringer** [comment]
@geigerj Did you have a chance to look at this?

**Comment by geigerj** [comment]
Ah, sorry. Just a quick question about `absolute_import`, otherwise looks good.

File: google/gax/utils/oneof.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Utility function to error if multiple "one of" arguments are sent."""
+
```
Comment (lukesneeringer) [inline google/gax/utils/oneof.py]: Possibly nothing, although the linter did pass.
That should definitely be added; I will do so and then merge, assuming everything else looks good to you.

**Comment by lukesneeringer** [comment]
Added. Am I clear to merge once CI passes?

**Comment by geigerj** [comment]
@lukesneeringer Yup, go for it once CI passes.

**Pull request merged**